### PR TITLE
arch/arm: Remove FAR and CODE from common/ and arm*/ folder

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Style change only

## Impact
Minor

## Testing
Pass CI

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@pkarashchenko macOS CI always fail to:
" wget --quiet https://static.dev.sifive.com/dev-tools/riscv64-unknown-elf-gcc-8.3.0-2019.08.0-x86_64-apple-darwin.tar.gz"
Let's ignore this error temporarily?

**Comment by pkarashchenko** [comment]
Yeah. `Error code: SEC_ERROR_EXPIRED_CERTIFICATE`. Seems like SiFive have to renew certficates

File: arch/arm/include/arch.h [hunk 1]
```diff
@@ -65,7 +65,7 @@ do { \
     "\tmov %0, " PIC_REG_STRING "\n\t" \
     : "=r"(picbase) \
   ); \
-  *ppicbase = (FAR void*)picbase; \
+  *ppicbase = (void*)picbase; \
```
Comment (pkarashchenko) [inline arch/arm/include/arch.h:68]:
```suggestion
  *ppicbase = (void *)picbase; \
```

File: arch/arm/src/common/arm_backtrace_thumb.c [hunk 2]
```diff
@@ -184,13 +184,13 @@ static bool in_code_region(FAR void *pc)
 #ifdef CONFIG_MM_KASAN
 __attribute__((no_sanitize_address))
 #endif
-static FAR void *backtrace_push_internal(FAR void **psp,
-                                         FAR void **ppc)
+static void *backtrace_push_internal(void **psp,
+                                         void **ppc)
```
Comment (pkarashchenko) [inline arch/arm/src/common/arm_backtrace_thumb.c:188]:
```suggestion
static void *backtrace_push_internal(void **psp,
                                     void **ppc)
```

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
Maybe we can issue a migration to `https://static.dev.sifive.com/dev-tools/freedom-tools/v2020.12/riscv64-unknown-elf-toolchain-10.2.0-2020.12.8-x86_64-apple-darwin.tar.gz` when certs are getting renewed.

**Comment by pkarashchenko** [comment]
@xiaoxiang781216 do you consider we can add `--no-check-certificate` as a workaround to get CI running? Or just wait assuming that SiFive will renew certs soon?

**Comment by xiaoxiang781216** [comment]
> Maybe we can issue a migration to `https://static.dev.sifive.com/dev-tools/freedom-tools/v2020.12/riscv64-unknown-elf-toolchain-10.2.0-2020.12.8-x86_64-apple-darwin.tar.gz` when certs are getting renewed.

let's upgrade to the new version:)?

**Comment by pkarashchenko** [comment]
> > Maybe we can issue a migration to `https://static.dev.sifive.com/dev-tools/freedom-tools/v2020.12/riscv64-unknown-elf-toolchain-10.2.0-2020.12.8-x86_64-apple-darwin.tar.gz` when certs are getting renewed.
> 
> let's upgrade to the new version:)?

The new version link also suffers same certificate expiration now :)

File: arch/arm/include/arch.h (hunk 1 above)
Comment (xiaoxiang781216) [reply arch/arm/include/arch.h]: Done.

File: arch/arm/src/common/arm_backtrace_thumb.c (hunk 2 above)
Comment (xiaoxiang781216) [reply arch/arm/src/common/arm_backtrace_thumb.c]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
> > > Maybe we can issue a migration to `https://static.dev.sifive.com/dev-tools/freedom-tools/v2020.12/riscv64-unknown-elf-toolchain-10.2.0-2020.12.8-x86_64-apple-darwin.tar.gz` when certs are getting renewed.
> > 
> > 
> > let's upgrade to the new version:)?
> 
> The new version link also suffers same certificate expiration now :)

Ok, I added --no-check-certificate in #6094 to workaround it.

**Pull request merged**